From: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Date: Mon, 6 Jun 2016 17:25:33 +0300
Subject: PCI / PM: check all fields in pci_set_platform_pm()
References: bnc#993388,FATE#321732
Patch-Mainline: v4.8-rc1
Git-commit: 6ec39cf5cd6f77e3ff9ff1da75b758a47f939888

When assign new PCI platform PM operations check for all mandatory fields to
prevent NULL pointer dereference.

Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/pci/pci.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/pci.c b/drivers/pci/pci.c
index c1e82e3..8c39b4e 100644
--- a/drivers/pci/pci.c
+++ b/drivers/pci/pci.c
@@ -536,8 +536,8 @@ static struct pci_platform_pm_ops *pci_platform_pm;
 
 int pci_set_platform_pm(struct pci_platform_pm_ops *ops)
 {
-	if (!ops->is_manageable || !ops->set_state || !ops->choose_state
-	    || !ops->sleep_wake)
+	if (!ops->is_manageable || !ops->set_state || !ops->choose_state ||
+	    !ops->sleep_wake || !ops->run_wake || !ops->need_resume)
 		return -EINVAL;
 	pci_platform_pm = ops;
 	return 0;
-- 
1.8.5.6

